perfkitbenchmarker/providers/gcp/gcp_spanner.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing class for GCP's spanner instances.
+
+Instances can be created and deleted.
+"""
+
+import logging
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import resource
+from perfkitbenchmarker.providers.gcp import util
+
+FLAGS = flags.FLAGS
+
+
+DEFAULT_CLOUD_SPANNER_END_POINT = 'https://spanner.googleapis.com/'
+DEFAULT_CLOUD_SPANNER_CONFIG = 'regional-us-central1'
+
+
+class GcpSpannerInstance(resource.BaseResource):
+  """Object representing a GCP Spanner Instance.
+
+  The project and Cloud Spanner config must already exist. Instance and database
+  will be created and torn down before and after the test.
+
+  Attributes:
+    name:        Name of the instance to create.
+    description: Description of the instance.
+    nodes:       Number of nodes in the instance.
+    database:    Name of the database to create
+    ddl:         The schema of the database.
+    project:     The gcloud project to use. If None, use the default project.
+    config:      The Cloud Spanner config to use.
+    end_point:   The end point of the Cloud Spanner. If None, use the default
+                 end point.
+  """
+
+  def __init__(self, name, description, nodes, database, ddl,
+               project=None, config=DEFAULT_CLOUD_SPANNER_CONFIG,
+               end_point=DEFAULT_CLOUD_SPANNER_END_POINT):
+    super(GcpSpannerInstance, self).__init__()
+    self._nodes = nodes
+    self._name = name
+    self._description = description
+    self._config = config
+    self._database = database
+    self._ddl = ddl
+    self._end_point = end_point
+
+    # Cloud Spanner may not explicitly set the following common flags.
+    self.project = project if project is not None else util.GetDefaultProject()
+    self.zone = None
+
+    self._OverrideEndPoint()
```

Comment:
My understanding is that PKB only supports serial execution of benchmarks, so this won't cause race condition.